src/HostProgram.elm

```diff
@@ -1,38 +1,46 @@
-module HostProgram exposing (create, Model, Msg)
+module HostProgram exposing (Model, Msg, create)
 
 {-| The FrameRouter module is the Elm code that backs the frame-router custom element
 in the iframe-coordinator toolkit. It handles mapping URL routes to clients displayed
 in a child frame as well as message validation and routing within the parent application.
 
 This module is not currently designed for stand-alone use. You should instead use the
-custom elements defined in LINK_TO_JS_LIB to create seamless iframe applications
+custom elements defined in LINK\_TO\_JS\_LIB to create seamless iframe applications
 
 @docs createRouter
 
 -}
 
+import ClientMessage exposing (ClientMessage)
 import ClientRegistry exposing (Client, ClientRegistry)
+import CommonMessages exposing (Publication)
+import HostMessage exposing (HostMessage)
 import Html exposing (Attribute, Html)
 import Html.Attributes exposing (attribute)
-import Json.Decode as Decode exposing (decodeValue)
-import ClientMessage exposing (ClientMessage)
+import Html.Events exposing (on)
+import Json.Decode as Decode exposing (Decoder, decodeValue)
+import LabeledMessage
 import Navigation exposing (Location)
 import Path exposing (Path)
+import Set exposing (Set)
 
 
 {-| Create a program to handle routing. Takes an input port to listen to messages on.
-port binding is handled in the custom frame-router element in LINK_TO_JS_LIB_HERE
+port binding is handled in the custom frame-router element in LINK\_TO\_JS\_LIB\_HERE
 -}
-create : ((Decode.Value -> Msg) -> Sub Msg) -> Program Decode.Value Model Msg
-create inputPort =
+create :
+    { fromHost : (Decode.Value -> Msg) -> Sub Msg
+    , toHost : Decode.Value -> Cmd Msg
+    }
+    -> Program Decode.Value Model Msg
+create ports =
```

Comment:
The HostProgram now has more ways it needs to send/recieve data from JavaScript, so the port arguments it gets on creation changes.  This wouldn't be needed if this wasn't designed to eventually be published on the Elm package system to take advantage of enforced SemVer when we change the API.